pkg/api/v1alpha1/cluster_types.go

```diff
@@ -832,11 +832,48 @@ var validCiliumPolicyEnforcementModes = map[CiliumPolicyEnforcementMode]bool{
 	CiliumPolicyModeNever:   true,
 }
 
+// Reasons for the terminal failures while reconciling the Cluster object.
+const (
+	// MissingDependentObjectsReason reports that the Cluster is missing dependent objects.
+	MissingDependentObjectsReason = "MissingDependentObjects"
+
+	// ManagementClusterRefInvalidReason reports that the Cluster management cluster reference is invalid. This
+	// can whether if it does not exist or the cluster referenced is not a management cluster.
+	ManagementClusterRefInvalidReason = "ManagementClusterRefInvalid"
+
+	// ClusterSpecValidationFailedReason reports that the Cluster spec validation has failed.
+	ClusterSpecValidationFailedReason = "ClusterSpecValidationFailed"
+
+	// DatacenterConfigValidationFailedReason reports that the Cluster datacenterconfig validation has failed.
+	DatacenterConfigValidationFailedReason = "DatacenterConfigValidationFailed"
+
+	// MachineConfigValidationFailedReason reports that the Cluster machineconfig validation has failed.
+	MachineConfigValidationFailedReason = "MachineConfigValidationFailed"
+
+	// UnavailableControlPlaneIPReason reports that the Cluster controlPlaneIP is already in use.
+	UnavailableControlPlaneIPReason = "UnavailableControlPlaneIP"
+)
+
+// Reasons for the terminal failures while reconciling the Cluster object specific for Tinkerbell.
+const (
+	// HardwareValidationFailedReason reports that the hardware validation has failed.
+	HardwareValidationFailedReason = "HardwareValidationFailed"
+
+	// RufioMachineValidationFailedReason reports that the rufio machine validation has failed.
+	RufioMachineValidationFailedReason = "RufioMachineValidationFailed"
```

Comment:
Can we avoid having the Rufio name in our API?
I think that the name of an specific internal component shouldn't bleed into our public API. What if the name of the component changes?